lib/internal/callbackify.js

```diff
@@ -0,0 +1,31 @@
+'use strict';
+
+module.exports = callbackify;
+
+const promises = require('internal/promises');
+const BuiltinPromise = promises.Promise;
+
+function callbackify(fn) {
+  Object.defineProperty(inner, 'name', {
+    writable: false,
+    enumerable: false,
+    configurable: true,
+    value: fn.name + 'Callback'
+  });
+  return inner;
+
+  function inner() {
+    const cb = arguments[arguments.length - 1];
+    if (typeof cb !== 'function') {
+      cb = (err) => { throw err; };
+    }
+    try {
+      BuiltinPromise.resolve(fn.apply(this, arguments)).then(
+        (result) => cb(null, result),
```

Comment:
This seems wrong to me - everything inside a callback passed to `then()` will be run inside of a `try`/`catch`. So if an error is thrown from `cb(null, result)` it will be caught by the promise implementation, and then swallowed because the promise is not returned, and has no `.catch()` at the end of the chain of calls.

Or are we saying that any promise implementation which can be plugged in will throw on any unhandled error? Hence the `catch` on line 27?